### PR TITLE
fix: [SRM-14256]: Update SLO page to reset filters whenever user changes the project

### PR DESCRIPTION
**Pull request by praneshg239**
### Summary

1. Reset all the filters in SLOs page, when user changes project
2. Reset page number when user changes project
3. Update to make fetch call happen only once
4. Added text filter for risk count

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by praneshg239** [comment]
🚨 Test case for updating project has not been added yet, I am yet to check the project change in unit test scenario. I will add the test case once I get it clarified.

File: cypress/integration/85-cv/slos/CreateCompositeSLO.spec.ts [hunk 1]
```diff
@@ -202,7 +202,7 @@ describe('Create account level SLO', () => {
     cy.contains('span', 'Service Reliability').click()
     cy.intercept('GET', getAccountLevelUserJourneysCall, listUserJourneysCallResponse)
   })
-  it('should be able to create SLO by filling all the details.', () => {
+  it.only('should be able to create SLO by filling all the details.', () => {
```
Comment (deepesh-ui) [inline cypress/integration/85-cv/slos/CreateCompositeSLO.spec.ts:205]: remove only

File: cypress/integration/85-cv/slos/CreateCompositeSLO.spec.ts [hunk 2]
```diff
@@ -251,6 +251,8 @@ describe('Create account level SLO', () => {
     cy.contains('p', 'demo_composite_slo').should('not.exist')
     cy.contains('span', 'Add SLOs').click()
     cy.wait('@sloListPostResponseRolling7Days')
+    // It is passing only if we add wait to it, Deepesh will be checking this
+    cy.wait(2000)
```
Comment (deepesh-ui) [inline cypress/integration/85-cv/slos/CreateCompositeSLO.spec.ts:255]: we have 2000 wait on below line as well

File: src/modules/85-cv/pages/slos/CVSLOsListingPage.tsx [hunk 3]
```diff
@@ -139,6 +145,45 @@ const CVSLOsListingPage: React.FC<CVSLOsListingPageProps> = ({ monitoredService
     queryParams: pathParams
   })
 
+  const fetchWidgetAndRiskCount = (queryParams?: Pick<GetSLOHealthListViewQueryParams, 'pageNumber'>): void => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/slos/CVSLOsListingPage.tsx:148]: Shall we move this function to utils

Comment (praneshg239) [reply src/modules/85-cv/pages/slos/CVSLOsListingPage.tsx:148]: `refetchDashboardWidgets` and `refetchRiskCount` are being used inside it. If we move the function to utils, then we have to pass it as a parameter every time. 

Just a heads up, as we are not sending it as a prop for other components, we don't need memoize as well.

**Review by deepesh-ui — APPROVED** [review]

**Review by karansaraswat19 — APPROVED** [review]

**Pull request merged**